s3_tar_ball.go

```diff
@@ -33,14 +31,14 @@ type S3TarBall struct {
 // SetUp creates a new tar writer and starts upload to S3.
 // Upload will block until the tar file is finished writing.
 // If a name for the file is not given, default name is of
-// the form `part_....tar.lz4`.
+// the form `part_....tar.[Compressor file extension]`.
 func (tarBall *S3TarBall) SetUp(crypter Crypter, names ...string) {
 	if tarBall.tarWriter == nil {
 		var name string
 		if len(names) > 0 {
 			name = names[0]
 		} else {
-			name = "part_" + fmt.Sprintf("%0.3d", tarBall.partCount) + ".tar." + Lz4FileExtension
+			name = "part_" + fmt.Sprintf("%0.3d", tarBall.partCount) + ".tar." + tarBall.tarUploader.compressor.FileExtension()
```

Comment:
How about tarFileExtension? If it's not overkill

Reply:
Fixed